src/imp.rs

```diff
@@ -632,3 +696,81 @@ fn generate_obfuscated_note<Rng: RngCore + CryptoRng>(
         blinder,
     )
 }
+
+#[cfg(test)]
+mod tests {
+    use rand::rngs::StdRng;
+    use rand_core::SeedableRng;
+
+    use super::*;
+
+    fn gen_notes(values: &[u64]) -> Vec<(Note, u64, JubJubScalar)> {
+        let mut rng = StdRng::seed_from_u64(0xbeef);
+
+        let ssk = SecretSpendKey::random(&mut rng);
+        let psk = ssk.public_spend_key();
+
+        let mut notes_and_values = Vec::with_capacity(values.len());
+
+        for value in values {
+            let note = Note::transparent(&mut rng, &psk, *value);
+            let blinder = JubJubScalar::random(&mut rng);
+
+            notes_and_values.push((note, *value, blinder));
+        }
+
+        notes_and_values
+    }
+
+    #[test]
+    fn note_picking_1() {
+        let values = [10];
+
+        let notes_and_values = gen_notes(&values);
+
+        let picked = pick_notes(10, notes_and_values);
+        assert_eq!(picked.len(), 1);
+    }
+
+    #[test]
+    fn note_picking_2() {
+        let values = [10, 20];
+
+        let notes_and_values = gen_notes(&values);
+
+        let picked = pick_notes(10, notes_and_values);
+        assert_eq!(picked.len(), 2);
+    }
+
+    #[test]
+    fn note_picking_3() {
+        let values = [10, 30, 20];
+
+        let notes_and_values = gen_notes(&values);
+
+        let picked = pick_notes(10, notes_and_values);
+        assert_eq!(picked.len(), 3);
+    }
+
+    #[test]
+    fn note_picking_4() {
+        let values = [40, 20, 10, 30];
+
+        let notes_and_values = gen_notes(&values);
+
+        let picked = pick_notes(10, notes_and_values);
+        assert_eq!(picked.len(), 4);
+    }
+
+    #[test]
+    fn note_picking_4_plus() {
+        let values = [10, 20, 30, 40, 50, 60, 70];
```

Comment:
Make it harder for the `pick_notes` to work out the right notes by passing these non sorted